_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp

```diff
@@ -5090,20 +5090,104 @@ mfxU64 make_back_color_yuv(mfxU16 bit_depth, mfxU16 Y, mfxU16 U, mfxU16 V)
+           ((mfxU64) VPP_RANGE_CLIP(B, 0, 255) << 0);
+};
+
+inline
+mfxU64 make_def_back_color_yuv(mfxU16 bit_depth)
+{
+    // A = max, Y=min, U,V = max/2
+    assert(bit_depth >= 8);
     mfxU64 const shift = bit_depth - 8;
-    mfxU64 const min_val = 16 << shift, max_val = 256 << shift;
+    mfxU16 const min_val = 16 << shift, max_val = 256 << shift;
     return
-        ((max_val - 1ULL) << 48) |
-        ( min_val      << 32) |
-        ( max_val / 2  << 16) |
-        ( max_val / 2  <<  0)
+        ((mfxU64)(max_val - 1ULL) << 48) |
+        ((mfxU64)VPP_RANGE_CLIP(min_val    , (16 << shift), (235 << shift))     << 32) |
+        ((mfxU64)VPP_RANGE_CLIP(max_val / 2, (16 << shift), (240 << shift))     << 16) |
+        ((mfxU64)VPP_RANGE_CLIP(max_val / 2, (16 << shift), (240 << shift))     <<  0)
         ;
 };
 
+inline
+mfxU64 make_def_back_color_argb(mfxU16 bit_depth)
+{
+    // A = max, R,G,B = 0
+    assert(bit_depth >= 8);
+    mfxU64 const shift = bit_depth - 8;
+    mfxU64 const max_val = 256 << shift;
+    return ((mfxU64)(max_val - 1ULL) << 48);
+};
+
+static
+mfxU64 get_background_color(const mfxVideoParam & videoParam)
+{
+    mfxU64 back_color = make_def_back_color_argb(8);
+
+    switch (videoParam.vpp.Out.FourCC)
+    {
+        case MFX_FOURCC_NV12:
+        case MFX_FOURCC_YV12:
+        case MFX_FOURCC_NV16:
+        case MFX_FOURCC_YUY2:
+        case MFX_FOURCC_AYUV:
+            back_color = make_def_back_color_yuv(8);
+            break;
+        case MFX_FOURCC_P010:
+#if (MFX_VERSION >= 1027)
+        case MFX_FOURCC_Y210:
+        case MFX_FOURCC_Y410:
+#endif
+        case MFX_FOURCC_P210:
+            back_color = make_def_back_color_yuv(10);
+            break;
+        default:
+            break;
+    }
+
+    for (mfxU32 i = 0; i < videoParam.NumExtParam; i++)
+    {
+        if (videoParam.ExtParam[i]->BufferId == MFX_EXTBUFF_VPP_COMPOSITE)
+        {
+            mfxExtVPPComposite *extComp = (mfxExtVPPComposite *) videoParam.ExtParam[i];
+            switch (videoParam.vpp.Out.FourCC) {
+                case MFX_FOURCC_NV12:
+                case MFX_FOURCC_YV12:
+                case MFX_FOURCC_NV16:
+                case MFX_FOURCC_YUY2:
+                case MFX_FOURCC_AYUV:
+                    back_color = make_back_color_yuv(8, extComp->Y, extComp->U, extComp->V);
+                    break;
+                case MFX_FOURCC_P010:
+#if (MFX_VERSION >= 1027)
+                case MFX_FOURCC_Y210:
+                case MFX_FOURCC_Y410:
+#endif
+                case MFX_FOURCC_P210:
+                    back_color = make_back_color_yuv(10, extComp->Y, extComp->U, extComp->V);
+                    break;
+                case MFX_FOURCC_RGB4:
+                case MFX_FOURCC_BGR4:
+                    back_color = make_back_color_argb(8, extComp->R, extComp->G, extComp->B);
+                    break;
+                default:
+                    break;
+            }
+        }
```

Comment:
and again, I think you can break after you've found MFX_EXTBUFF_VPP_COMPOSITE :)

Reply:
Even more:
inside if{} in switch **return make_back_color** *.
First switch for default back_color move after the loop. And return each time immediately, don't use back_color at all.